rest_client.go

```diff
@@ -55,6 +55,25 @@ func handleError(resp *http.Response) error {
 	return errors.WithStack(gimerr)
 }
 
+func (c *restClient) doRequest(ctx context.Context, method string, url string, body io.Reader) (*http.Response, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem building request")
+	}
+
+	req = req.WithContext(ctx)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem making request")
+	}
+	if err = handleError(resp); err != nil {
+		defer resp.Body.Close()
```

Comment:
Why only close in the error case?

Reply:
I assumed that the successful server response status was 200 ok. Returning an error on line 71 means the server returned non-200 so the caller of `doRequest` is not responsible for closing the response body since the server did not successfully perform the operation. Returning nil error means the server returned 200 and the caller can use the response object (so they're also responsible for closing the response).

At the moment, the user is only responsible for closing `resp.Body` if there was no error. If it's cleaner, I could just move the handleError portion out of this function and then the user would always be responsible for closing the response body.

Reply:
Ah, gotcha, that makes sense. LGTM.